src/tests.rs

```diff
@@ -0,0 +1,510 @@
+        let add_schema_set = BasePrincipalSet::from(vec![BasePrincipal::GroupMember(0)]);
+        assert_ok!(Permissions::set_class_add_schemas_set(
+            Origin::ROOT,
+            None,
+            class_id,
+            add_schema_set
+        ));
+
+        // successfully add a new schema
+        assert_ok!(Permissions::add_class_schema(
+            Origin::signed(MEMBER_ONE_OF_GROUP_ZERO),
+            Some(0),
+            class_id,
+            vec![],
+            simple_test_schema()
+        ));
+
+        // System can always create entities (provided class exists) bypassing any permissions
+        let entity_id_1 = <versioned_store::Module<Runtime>>::next_entity_id();
+        assert_ok!(Permissions::create_entity(Origin::ROOT, None, class_id,));
+        // entities created by system are "un-owned"
+        assert!(!<EntityOwnerByEntityId<Runtime>>::exists(entity_id_1));
+        assert_eq!(Permissions::entity_owner_by_entity_id(entity_id_1), None);
+
+        // default permissions have empty create_entities set and by default no entities can be created
+        assert_err!(
+            Permissions::create_entity(Origin::signed(MEMBER_ONE_OF_GROUP_ONE), Some(1), class_id,),
+            "EntitiesCannotBeCreated"
+        );
+
+        assert_ok!(Permissions::set_class_entities_can_be_created(
+            Origin::ROOT,
+            None,
+            class_id,
+            true
+        ));
+
+        assert_err!(
+            Permissions::create_entity(Origin::signed(MEMBER_ONE_OF_GROUP_ONE), Some(1), class_id,),
+            "NotInCreateEntitiesSet"
+        );
+
+        // give members of GROUP_ONE permission to create entities
+        let create_entities_set = BasePrincipalSet::from(vec![BasePrincipal::GroupMember(1)]);
+        assert_ok!(Permissions::set_class_create_entities_set(
+            Origin::ROOT,
+            None,
+            class_id,
+            create_entities_set
+        ));
+
+        let entity_id_2 = <versioned_store::Module<Runtime>>::next_entity_id();
+        assert_ok!(Permissions::create_entity(
+            Origin::signed(MEMBER_ONE_OF_GROUP_ONE),
+            Some(1),
+            class_id,
+        ));
+        assert!(<EntityOwnerByEntityId<Runtime>>::exists(entity_id_2));
+        assert_eq!(
+            Permissions::entity_owner_by_entity_id(entity_id_2),
+            Some(BasePrincipal::GroupMember(1))
+        );
+
+        // Updating entity must be authorized
+        assert_err!(
+            Permissions::add_schema_support_to_entity(
+                Origin::signed(1),
+                None,
+                false, // not claiming to be entity owner
+                entity_id_2,
+                0, // first schema created
+                simple_test_entity_property_values()
+            ),
+            "NotInEntityPermissionsUpdateSet"
+        );
+
+        // default permissions give entity owner permission to update and delete
```

Comment:
Versioned store doesn't have a functionality to delete something on Entity.
```suggestion
        // default permissions give entity owner permission to update
```